fastapi/models/fastapi_endpoint_demo.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2022 ACSONE SA/NV
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL).
+
+from enum import Enum
+from typing import List
+
+from odoo import _, api, fields, models
+from odoo.api import Environment
+from odoo.exceptions import AccessError, MissingError, UserError, ValidationError
+
+from odoo.addons.base.models.res_partner import Partner
+
+from fastapi import APIRouter, Depends, HTTPException, status
+from fastapi.security import APIKeyHeader
+from pydantic import BaseModel, Field
+
+from ..depends import (
+    authenticated_partner,
+    authenticated_partner_from_basic_auth_user,
+    authenticated_partner_impl,
+    fastapi_endpoint,
+    odoo_env,
+)
+
+
+class FastapiEndpoint(models.Model):
+
+    _inherit = "fastapi.endpoint"
+
+    app: str = fields.Selection(
+        selection_add=[("demo", "Demo Endpoint")], ondelete={"demo": "cascade"}
+    )
+    demo_auth_method = fields.Selection(
+        selection=[("api_key", "Api Key"), ("http_basic", "HTTP Bacic")],
+        string="Authenciation method",
+    )
+
+    def _get_fastapi_routers(self) -> List[APIRouter]:
+        if self.app == "demo":
+            return [demo_api_router]
+        return super()._get_fastapi_routers()
+
+    @api.constrains("app", "demo_auth_method")
+    def _valdiate_demo_auth_method(self):
+        for rec in self:
+            if rec.app == "demo" and not rec.demo_auth_method:
+                raise ValidationError(
+                    _(
+                        "The authentication method is required for app %(app)s",
+                        app=rec.app,
+                    )
+                )
+
+    @api.model
+    def _fastapi_app_fields(self) -> List[str]:
+        fields = super()._fastapi_app_fields()
+        fields.append("demo_auth_method")
+        return fields
+
+    def _get_app(self):
+        app = super()._get_app()
+        if self.app == "demo":
+            # Here we add the overrides to the authenticated_partner_impl method
+            # according to the authentication method configured on the demo app
+            if self.demo_auth_method == "http_basic":
+                authenticated_partner_impl_override = (
+                    authenticated_partner_from_basic_auth_user
+                )
+            else:
+                authenticated_partner_impl_override = (
+                    api_key_based_authenticated_partner_impl
+                )
+        app.dependency_overrides[
+            authenticated_partner_impl
+        ] = authenticated_partner_impl_override
```

Comment:
```suggestion
            app.dependency_overrides[
                authenticated_partner_impl
            ] = authenticated_partner_impl_override
```

Reply:
Indentation is wrong, so when self.app is different then "demo" it raise an error as authenticated_partner_impl_override is not defined